net/pbr/Makefile

```diff
@@ -0,0 +1,198 @@
+# Copyright 2017-2022 Stan Grishin (stangri@melmac.ca)
+# This is free software, licensed under the GNU General Public License v3.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=pbr
+PKG_VERSION:=0.9.9
+PKG_RELEASE:=24
+PKG_LICENSE:=GPL-3.0-or-later
+PKG_MAINTAINER:=Stan Grishin <stangri@melmac.ca>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/pbr/default
+	SECTION:=net
+	CATEGORY:=Network
+	SUBMENU:=VPN
+	PROVIDES:=pbr
+	TITLE:=Policy Based Routing Service
+	URL:=https://docs.openwrt.melmac.net/pbr/
+	DEPENDS:=+ip-full +jshn +jsonfilter +resolveip
+	CONFLICTS:=vpnbypass vpn-policy-routing
+	PKGARCH:=all
+endef
+
+define Package/pbr
+$(call Package/pbr/default)
+	TITLE+= with nft/nft set support
+	DEPENDS+=+firewall4 +kmod-nft-core +kmod-nft-nat +nftables-json
+endef
+
+define Package/pbr-iptables
+$(call Package/pbr/default)
+	TITLE+= with iptables/ipset support
+	DEPENDS+=+ipset +iptables +kmod-ipt-ipset +iptables-mod-ipopt
+endef
+
+define Package/pbr-netifd
+$(call Package/pbr/default)
+	TITLE+= with netifd support
+endef
+
+define Package/pbr/description
+This service enables policy-based routing for WAN interfaces and various VPN tunnels.
+This version supports OpenWrt with both fw3/ipset/iptables and fw4/nft.
+endef
+
+define Package/pbr-iptables/description
+This service enables policy-based routing for WAN interfaces and various VPN tunnels.
+This version supports OpenWrt with fw3/ipset/iptables.
+endef
+
+define Package/pbr-netifd/description
+This service enables policy-based routing for WAN interfaces and various VPN tunnels.
+This version supports OpenWrt with both fw3/ipset/iptables and fw4/nft.
+This version uses OpenWrt native netifd/tables to set up interfaces. This is WIP.
+endef
+
+define Package/pbr/default/conffiles
+/etc/config/pbr
+endef
+
+Package/pbr/conffiles = $(Package/pbr/default/conffiles)
+Package/pbr-iptables/conffiles = $(Package/pbr/default/conffiles)
+Package/pbr-netifd/conffiles = $(Package/pbr/default/conffiles)
+
+define Build/Configure
+endef
+
+define Build/Compile
+endef
+
+define Package/pbr/default/install
+	$(INSTALL_DIR) $(1)/etc/config
```

Comment:
What rather to create directory first and then what files should be in that directory and etc? Right now, it is not very clear unless going through into it deeply what goes where.

Reply:
Sorry, I'm not sure I'm getting your point, could you rephrase it?

Reply:
I think the idea was to group `INSTALL_DIR` with respective `INSTALL_DATA`/`BIN` so that it is clear which files go to which directories. Meaning to have something like this (like it is in `package/libs/libselinux/Makefile` for example):

```bash
$(INSTALL_DIR) $(1)/etc/init.d
$(INSTALL_BIN) ./files/etc/init.d/pbr.init $(1)/etc/init.d/pbr
$(SED) "s|^\(readonly PKG_VERSION\).*|\1='$(PKG_VERSION)-$(PKG_RELEASE)'|" $(1)/etc/init.d/pbr
$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
$(INSTALL_DATA) ./files/etc/hotplug.d/iface/70-pbr $(1)/etc/hotplug.d/iface/70-pbr
…
```

This way it will be visible which directories are created empty for example.

Reply:
Hey @oldium, thanks! This is exactly what I meant.

Reply:
Thanks, I've reorganized that in 0.9.9-37 and will include this in the next push.